### PR TITLE
Yechezkel/omer jooq

### PR DESCRIPTION
**Pull request by chezi-nikop**
finish with creating fun and tests for owner and pet

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/dao/PetsDao.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            .where(pet.type.eq(getType(type)))
+            .fetch(petMapper)
+    }
+
+    fun getType(type: PetType): String {
+        return when (type) {
+            PetType.DOG -> "DOG"
+            PetType.CAT -> "CAT"
+        }
+    }
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:36]: Move to the Enum implementation, you can use this function in all over the system so it isnt smart to put it here.

File: src/main/kotlin/com/hibob/academy/dao/PetsDao.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    fun createPet(newPetData: PetData) : Long? {
+        return sql.insertInto(pet)
+            .set(pet.ownerId, newPetData.ownerId)
+            .set(pet.name, newPetData.name)
+            .set(pet.type, newPetData.type)
+            .set(pet.companyId, newPetData.companyId)
+            .set(pet.dateOfArrival, newPetData.dateOfArrival)
+            .returning(pet.petId)
+            .fetchOne()?.let { it[pet.petId] }
+    }
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:47]: Why not use !! ?

File: src/main/kotlin/com/hibob/academy/dao/PetsDao.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    fun getPet(petId: Long?): PetData? {
+        return sql.select(pet.ownerId, pet.petId, pet.name, pet.type, pet.companyId, pet.dateOfArrival)
+            .from(pet)
+            .where(pet.petId.eq(petId))
+            .fetchOneInto(PetData::class.java)
+    }
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:54]: Do you think of a scenario when you query by null?
Moreover, add companyId here also.

File: src/main/kotlin/com/hibob/academy/dao/PetsDao.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    fun  updateThePetOwnerWithTheOwnerId(petId: Long?, ownerId: Long) {
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:56]: Im sure you can think of a better name for this function

Comment (omermelamed) [reply src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:56]: Same here, why nullable id?

Comment (omermelamed) [reply src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:56]: Add companyId as well in the where like all other functions in the dao.

File: src/main/kotlin/com/hibob/academy/dao/models.kt [hunk 5]
```diff
@@ -1,3 +1,9 @@
...
 
-data class Example(val id: Long, val companyId: Long, val data: String)
\ No newline at end of file
+import java.time.LocalDate
+
+data class Example(val id: Long, val companyId: Long, val data: String)
+
+data class PetData(val ownerId: Long? , val petId: Long, val name: String, val type: String, val companyId: Long, val dateOfArrival: LocalDate)
+
+data class OwnerData(val name: String, val companyId: Long, val employeeId: String)
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/models.kt:9]: If pet holds ownerId so why doesnt owner holds id?

File: src/main/kotlin/com/hibob/academy/dao/tables.kt [hunk 6]
```diff
@@ -0,0 +1,32 @@
+package com.hibob.academy.dao
+
+import com.hibob.academy.utils.JooqTable
+
+class Pets(tableName: String = "pets") : JooqTable(tableName) {
+    val ownerId = createBigIntField("owner_id")
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/tables.kt:6]: Id should be the first column, ownerid is a reference for id in other table, please move it to be after the table id (petId in this case)

File: src/main/resources/db/migration/V202409091400__create_pets_table.sql [hunk 7]
```diff
@@ -1,6 +1,6 @@
 create table Pets
 (
-    id SERIAL primary key,
+    id BIGSERIAL primary key,
```
Comment (omermelamed) [inline src/main/resources/db/migration/V202409091400__create_pets_table.sql:3]: Please remove all migration code from this pr and put them in a different pr that contains only migrations.

File: src/main/resources/db/migration/V202409091457__create_owner_table.sql [hunk 8]
```diff
@@ -2,9 +2,9 @@ create table Owner
 (
     id SERIAL primary key,
     name varchar(255) not null,
-    company_id int not null,
+    company_id bigint not null,
```
Comment (omermelamed) [inline src/main/resources/db/migration/V202409091457__create_owner_table.sql:5]: Please remove all migration code from this pr and put them in a different pr that contains only migrations

File: src/main/resources/db/migration/V202409091457__create_owner_table.sql [hunk 9]
```diff
@@ -2,9 +2,9 @@ create table Owner
...
     employee_id varchar(255) not null
 );
 
-CREATE INDEX idx_owner_company_id_employee_id on Owner(company_id, employee_id);
+CREATE UNIQUE INDEX idx_owner_company_id_employee_id on Owner(company_id, employee_id);
```
Comment (omermelamed) [inline src/main/resources/db/migration/V202409091457__create_owner_table.sql:9]: Please remove all migration code from this pr and put them in a different pr that contains only migrations

File: src/main/resources/db/migration/V202409111715__add_owner_to_pets_table.sql [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+--where type = 'dog';
+
+--DELETE FROM Pets where id = '2';
+
+--select *
+--from Pets
+--where date_of_arrival <= '2023-09-09';
+
+
+
```
Comment (omermelamed) [inline src/main/resources/db/migration/V202409111715__add_owner_to_pets_table.sql:22]: Please remove all migration code from this pr and put them in a different pr that contains only migrations

File: src/main/resources/db/migration/V202409121600__make_company_id_bigint_on_pets_table.sql [hunk 11]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE Pets
+ALTER COLUMN company_id TYPE BIGINT;
```
Comment (omermelamed) [inline src/main/resources/db/migration/V202409121600__make_company_id_bigint_on_pets_table.sql:2]: Please remove all migration code from this pr and put them in a different pr that contains only migrations

**Review by omermelamed — COMMENTED** [review]
Few comments.
You are missing a function that get the owner data by petId.
I didn't go over the tests yet.

**Review by omermelamed — COMMENTED** [review]
Nice 👌🏻 you are missing a function that return OwnerData from petId and companyId.
Please add it and send me ti review.

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+    fun createOwnerIfNotExists(newOwnerData: OwnerDataInsert): Long? {
+        return  sql.insertInto(owner)
+            .set(owner.name, newOwnerData.name)
+            .set(owner.companyId, newOwnerData.companyId)
+            .set(owner.employeeId, newOwnerData.employeeId)
+            .onConflict(owner.companyId, owner.employeeId)
+            .doNothing()
+            .returning(owner.ownerId)
+            .fetchOne()?.let { it[owner.ownerId] }
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt:36]: Why not !! ?

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+    fun getOwnerById(id: Long, companyId: Long): OwnerData? {
+        return sql.select(owner.ownerId, owner.name, owner.companyId, owner.employeeId)
+            .from(owner)
+            .where(owner.ownerId.equal(id), owner.companyId.equal(companyId))
+            .fetchOneInto(OwnerData::class.java)
+    }
+
+    fun getOwnerByPetId(petId: Long, companyId: Long): OwnerData? {
+        val petsTable = PetsTable()
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt:47]: Please define the table at the beginning of the class

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+        return sql.select(owner.ownerId, owner.name, owner.companyId, owner.employeeId)
+            .from(owner)
+            .join(petsTable).on(petsTable.ownerId.eq(owner.ownerId))
+            .where(petsTable.petId.eq(petId), petsTable.companyId.eq(companyId))
+            .fetchOneInto(OwnerData::class.java)
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt:52]: Why not use the mapper?

File: src/main/kotlin/com/hibob/academy/dao/PetsDao.kt [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+            .set(pet.companyId, newPetData.companyId)
+            .returning(pet.petId)
+            .fetchOne()!![pet.petId]
+    }
+
+    fun getPet(petId: Long, companyId: Long): PetData? {
+        return sql.select(pet.ownerId, pet.petId, pet.name, pet.type, pet.companyId, pet.dateOfArrival)
+            .from(pet)
+            .where(pet.petId.eq(petId), pet.companyId.eq(companyId))
+            .fetchOneInto(PetData::class.java)
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:44]: Why not use the mapper?

File: src/main/kotlin/com/hibob/academy/dao/models.kt [hunk 16]
```diff
@@ -1,3 +1,14 @@
 package com.hibob.academy.dao
 
-data class Example(val id: Long, val companyId: Long, val data: String)
\ No newline at end of file
+import java.time.LocalDate
+
+data class Example(val id: Long, val companyId: Long, val data: String)
+
+data class PetData( val petId: Long, val ownerId: Long?, val name: String, val type: String, val companyId: Long, val dateOfArrival: LocalDate)
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/models.kt:7]: type should be from type PetType right?

File: src/main/kotlin/com/hibob/academy/dao/models.kt [hunk 17]
```diff
@@ -1,3 +1,14 @@
...
+
+data class PetDataInsert(val ownerId: Long?, val name: String, val type: String, val companyId: Long)
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/models.kt:9]: Same

File: src/main/kotlin/com/hibob/academy/dao/PetsDao.kt [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+
+    private val pet = PetsTable.instance
+
+    private val petMapper = RecordMapper<Record, PetData>
+    { record ->
+        PetData(
+            record[pet.petId],
+            record[pet.ownerId],
+            record[pet.name],
+            record[pet.type],
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:17]: I think you should map the type to PetType type

File: src/main/resources/db/migration/V202409091400__create_pets_table.sql (hunk 7 above)
Comment (omermelamed) [reply src/main/resources/db/migration/V202409091400__create_pets_table.sql:3]: Please remove migration code

**Review by omermelamed — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt (hunk 12 above)
Comment (chezi-nikop) [reply src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt:36]: If there is already an owner I want it to return null

Comment (omermelamed) [reply src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt:37]: Why not throw an exception?

File: src/main/kotlin/com/hibob/academy/dao/PetsDao.kt [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+            enumValueOf<PetType>(record[pet.type]),
+            record[pet.companyId],
+            record[pet.dateOfArrival]
+        )
+    }
+
+    fun getAllPetsByType(type: PetType, companyId: Long) : List<PetData> {
+        return sql.select(pet.ownerId, pet.petId, pet.name, pet.type, pet.companyId, pet.dateOfArrival)
+            .from(pet)
+            .where(pet.type.eq(typeEnumToString(type)), pet.companyId.eq(companyId))
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:26]: why not use .and here?

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+            .from(owner)
+            .where(owner.ownerId.equal(id), owner.companyId.equal(companyId))
+            .fetchOne(ownerMapper)
+    }
+
+    fun getOwnerByPetId(petId: Long, companyId: Long): OwnerData? {
+        return sql.select(owner.ownerId, owner.name, owner.companyId, owner.employeeId)
+            .from(owner)
+            .join(petsTable).on(petsTable.ownerId.eq(owner.ownerId))
+            .where(petsTable.petId.eq(petId), petsTable.companyId.eq(companyId))
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt:51]: why not use .and here?

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+            .onConflict(owner.companyId, owner.employeeId)
+            .doNothing()
+            .returning(owner.ownerId)
+            .fetchOne()?.let { it[owner.ownerId] }
+    }
+
+    fun getOwnerById(id: Long, companyId: Long): OwnerData? {
+        return sql.select(owner.ownerId, owner.name, owner.companyId, owner.employeeId)
+            .from(owner)
+            .where(owner.ownerId.equal(id), owner.companyId.equal(companyId))
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt:43]: why not use .and here?

File: src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt [hunk 22]
```diff
@@ -0,0 +1,119 @@
...
+        val petTest = PetDataInsert(insertOwner.ownerId , name = "A", PetType.DOG , companyId)
+        val newPetId = petDao.createPet(petTest)
+
+        val checkOwnerTest = ownerDao.getOwnerByPetId(newPetId, companyId)
+
+        assertNotNull(checkOwnerTest)
+        assertEquals(insertOwner.name, checkOwnerTest?.name)
+        assertEquals(insertOwner.employeeId, checkOwnerTest?.employeeId)
+        assertEquals(insertOwner.companyId, checkOwnerTest?.companyId)
+    }
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt:100]: Why not compare Owner to Owner?
try to compare the checkOwnerTest to a known Owner

File: src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt [hunk 23]
```diff
@@ -0,0 +1,119 @@
...
+
+    @Test
+    fun `create a new owner that exist in the database`() {
+        val newOwner = OwnerDataInsert(name, companyId, employeeId = "1")
+
+        ownerDao.createOwnerIfNotExists(newOwner)
+        ownerDao.createOwnerIfNotExists(newOwner)
+
+        assertEquals(1, ownerDao.getAllOwners(companyId).size)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt:57]: compare the returned list to a known list

File: src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt [hunk 24]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    @Test
+    fun `create a new owner that doesn't exist in the database`() {
+        val newOwner = OwnerDataInsert(name, companyId, employeeId = "1")
+        ownerDao.createOwnerIfNotExists(newOwner)
+
+        assertTrue(newOwner.name.equals(ownerDao.getAllOwners(companyId)[0].name))
+        assertTrue(newOwner.companyId.equals(ownerDao.getAllOwners(companyId)[0].companyId))
+        assertTrue(newOwner.employeeId.equals(ownerDao.getAllOwners(companyId)[0].employeeId))
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt:47]: Why not compare List<Owner> to List<Owner>?
try to compare the owners that returned from the function to a known Owners

File: src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt [hunk 25]
```diff
@@ -0,0 +1,119 @@
...
+
+    @Test
+    fun `get all owners wen we got owners in the database`() {
+        val newOwner1 = OwnerDataInsert( name, companyId, employeeId = "1")
+        val newOwner2 = OwnerDataInsert( name, companyId, employeeId = "2")
+
+        ownerDao.createOwnerIfNotExists(newOwner1)
+        ownerDao.createOwnerIfNotExists(newOwner2)
+
+        assertEquals(2, ownerDao.getAllOwners(companyId).size)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/OwnerDaoTest.kt:32]: Why not compare List to List?
try to compare the owners that returned from the function to a known Owners

File: src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt [hunk 26]
```diff
@@ -0,0 +1,106 @@
...
+        val petId2 = 1L
+        val petTest1 = PetDataInsert(petId1, name = "A", PetType.DOG, companyId)
+        val petTest2 = PetDataInsert(petId2, name = "B", PetType.DOG, companyId)
+
+        petDao.createPet(petTest1)
+        petDao.createPet(petTest2)
+
+        val allPetsInData = petDao.getAllPetsByType(PetType.DOG, companyId)
+
+        assertEquals(2, allPetsInData.size)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt:38]: Why not compare List to List?
try to compare the pets that returned from the function to a known Pets

File: src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt [hunk 27]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    @Test
+    fun `get all pets by type wen we dont have pets in the data base`() {
+
+        val allPetsInData = petDao.getAllPetsByType(PetType.DOG, companyId)
+
+        assertEquals(0, allPetsInData.size)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt:46]: List.empty = allPetsInData is better

File: src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt [hunk 28]
```diff
@@ -0,0 +1,106 @@
...
+
+    @Test
+    fun `create a new pet that doesn't exist in the database`() {
+        val petTest = PetDataInsert(ownerId, name = "A", PetType.DOG, companyId)
+
+        petDao.createPet(petTest)
+
+        val filteredPets = petDao.getAllPetsByType(PetType.DOG, companyId)
+
+        assertEquals(1, filteredPets.size)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt:57]: Why not compare List to List?
try to compare the pets that returned from the function to a known Pets

File: src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt [hunk 29]
```diff
@@ -0,0 +1,106 @@
...
+    @Test
+    fun `return the pet using its id when it exists in the database`() {
+        val petTest = PetDataInsert(ownerId, name = "A", PetType.DOG, companyId)
+
+        val newPetId = petDao.createPet(petTest)
+
+        assertEquals(petTest.ownerId, petDao.getPet(newPetId, companyId)?.ownerId)
+        assertEquals(petTest.name, petDao.getPet(newPetId, companyId)?.name)
+        assertEquals(petTest.type, petDao.getPet(newPetId, companyId)?.type)
+        assertEquals(petTest.companyId, petDao.getPet(newPetId, companyId)?.companyId)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt:70]: Why not compare Pet to Pet?
try to compare the pet that returned from the function to a known Pet

File: src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt [hunk 30]
```diff
@@ -0,0 +1,106 @@
...
+    @Test
+    fun `update the pet ownerId wen the ownerId exist`() {
+        val petTest = PetDataInsert(ownerId, name = "A", PetType.DOG, companyId)
+        val newPetId = petDao.createPet(petTest)
+
+        val newOwnerId = 2L
+
+        petDao.updatePetOwnerId(newPetId, newOwnerId, companyId)
+
+        assertEquals(ownerId, petDao.getPet(newPetId, companyId)?.ownerId)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt:92]: Why not compare Pet to List?
try to compare the pet that returned from the function to a known Pet

File: src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt [hunk 31]
```diff
@@ -0,0 +1,106 @@
...
+    @Test
+    fun `update the pet owner with the owner id wen the ownerId is null`() {
+        val petTest = PetDataInsert(null, name = "A", PetType.DOG, companyId)
+        val newPetId = petDao.createPet(petTest)
+
+        val newOwnerId = 2L
+
+        petDao.updatePetOwnerId(newPetId, newOwnerId, companyId)
+
+        assertEquals(newOwnerId, petDao.getPet(newPetId, companyId)?.ownerId)
```
Comment (omermelamed) [inline src/test/kotlin/com/hibob/academy/dao/PetsDaoTest.kt:104]: Why not compare Pet to Pet?
try to compare the pet that returned from the function to a known Pet

File: src/main/kotlin/com/hibob/academy/service/ExampleService.kt
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/service/ExampleService.kt:1]: @omermelamed ignore this file, this is related to fixes that @chezi-nikop and i needed to make to work out his github

File: src/main/kotlin/com/hibob/kotlinBasic/stringIsValid.kt
Comment (elijulian) [inline src/main/kotlin/com/hibob/kotlinBasic/stringIsValid.kt:1]: @omermelamed this one too

File: src/main/kotlin/com/hibob/academy/dao/PetsDao.kt [hunk 32]
```diff
@@ -0,0 +1,58 @@
...
+            record[pet.ownerId],
+            record[pet.name],
+            enumValueOf<PetType>(record[pet.type]),
+            record[pet.companyId],
+            record[pet.dateOfArrival]
+        )
+    }
+
+    fun getAllPetsByType(type: PetType, companyId: Long) : List<PetData> {
+        return sql.select(pet.ownerId, pet.petId, pet.name, pet.type, pet.companyId, pet.dateOfArrival)
```
Comment (chezi-nikop) [inline src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:24]: Its Omer - you mixes between the ownerId and PetId

File: src/main/kotlin/com/hibob/academy/dao/PetsDao.kt [hunk 33]
```diff
@@ -0,0 +1,58 @@
...
+            .set(pet.ownerId, newPetData.ownerId)
+            .set(pet.name, newPetData.name)
+            .set(pet.type, typeEnumToString(newPetData.type))
+            .set(pet.companyId, newPetData.companyId)
+            .returning(pet.petId)
+            .fetchOne()!![pet.petId]
+    }
+
+    fun getPet(petId: Long, companyId: Long): PetData? {
+        return sql.select(pet.ownerId, pet.petId, pet.name, pet.type, pet.companyId, pet.dateOfArrival)
```
Comment (chezi-nikop) [inline src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:42]: same here

File: src/main/kotlin/com/hibob/academy/dao/PetsDao.kt [hunk 34]
```diff
@@ -0,0 +1,58 @@
...
+
+    fun updatePetOwnerId(petId: Long, ownerId: Long, companyId: Long) : Long? {
+        return sql.update(pet)
+            .set(pet.ownerId, ownerId)
+            .where(pet.petId.eq(petId)
+            .and(pet.ownerId.isNull))
+            .and(pet.companyId.eq(companyId))
+            .returning(pet.petId)
+            .fetchOne()?.let { it[pet.petId] }
+    }
```
Comment (chezi-nikop) [inline src/main/kotlin/com/hibob/academy/dao/PetsDao.kt:57]: What does the id really mean here?

**Review by omermelamed — APPROVED** [review]

**Pull request merged**